Common/Data/UniverseSelection/BenchmarkUniverse.cs

```diff
@@ -0,0 +1,107 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+*/
+
+using System;
+using System.Collections.Generic;
+using System.Collections.Specialized;
+using System.Linq;
+using QuantConnect.Interfaces;
+using QuantConnect.Securities;
+
+namespace QuantConnect.Data.UniverseSelection
+{
+    /// <summary>
+    /// Represents the universe containing only the selected benchmark security.
+    /// This universe keeps the benchmark security and its subscription separate
+    /// from the algorithm securities. Universe selection will fire only once,
+    /// when the benchmark security is added to the universe.
+    /// </summary>
+    public class BenchmarkUniverse : Universe, INotifyCollectionChanged
+    {
+        /// <summary>
+        /// Event fired when a symbol is added or removed from this universe
+        /// </summary>
+        public event NotifyCollectionChangedEventHandler CollectionChanged;
+
+        /// <summary>
+        /// Gets the settings used for subscriptions added for this universe (unused in this universe)
+        /// </summary>
+        public override UniverseSettings UniverseSettings => null;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="BenchmarkUniverse"/> class
+        /// </summary>
+        /// <param name="configuration">The configuration used to resolve the data for universe selection</param>
+        public BenchmarkUniverse(SubscriptionDataConfig configuration) : base(configuration)
+        {
+        }
+
+        /// <summary>
+        /// Returns the benchmark symbol in this universe
+        /// </summary>
+        /// <param name="utcTime">The current UTC time</param>
+        /// <param name="data">The symbols to remain in the universe</param>
+        public override IEnumerable<Symbol> SelectSymbols(DateTime utcTime, BaseDataCollection data)
+        {
+            return Securities.Select(x => x.Key);
+        }
+
+        /// <summary>
+        /// Returns an existing security for this universe
+        /// </summary>
+        public override Security GetSecurity(Symbol symbol, IAlgorithm algorithm)
+        {
+            return Securities.Select(x => x.Value.Security).FirstOrDefault(x => x.Symbol == symbol);
+        }
+
+        /// <summary>
+        /// Adds the specified symbol to this universe
+        /// </summary>
+        /// <param name="utcTime">The current UTC time</param>
+        /// <param name="security">The symbol to be added to this universe</param>
+        /// <returns>True if the symbol was added, false if it was already present</returns>
+        public bool AddBenchmarkSecurity(DateTime utcTime, Security security)
+        {
+            if (AddMember(utcTime, security))
+            {
+                OnCollectionChanged(new NotifyCollectionChangedEventArgs(NotifyCollectionChangedAction.Add, security.Symbol));
```

Comment:
Awesome usage of `INotifyCollectionChanged` -- hooks right into the existing infrastructure set up for `UserDefinedUniverse` 👍